lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1546,17 +1610,59 @@ bool LinearMapInfo::shouldDifferentiateInstruction(SILInstruction *inst) {
       [&](SILValue val) { return activityInfo.isActive(val, indices); });
   if (hasActiveOperands && hasActiveResults)
     return true;
-  if (inst->mayHaveSideEffects() && hasActiveOperands)
-    return true;
+
+  // TODO: temporary hack while we investigate why reverse mode doesn't work
+  // with us handling `apply` instructions differently.
+  switch (kind) {
+  case AutoDiffLinearMapKind::Differential: {
+    CHECK_INST_TYPE_ACTIVE_DEST(StoreInst)
```

Comment:
It's better to define this kind of macros immediately before you use them and undefine immediately after.

Reply:
Done in https://github.com/apple/swift/pull/26743/commits/73f117604934b50ac600be3c9713fa39436d9d8a. Macro `#define`/`#undef` are tight around the use site.